File: Summarization/src/main/java/graph/spark/parser/SparkParser.java
package graph.spark.parser;

import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.function.Function;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

public class SparkParser
{
    public static <T> JavaRDD<T> parseJson( JavaRDD<String> input, Function<JsonNode, T> schematizer )
    {
        return input.map( string -> schematizer.call( ( ( new ObjectMapper() ).readTree( string ) ) ) );
    }
}

File: src/main/scala/graph/spark/parser/SparkParser.scala
package graph.spark.parser

import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
import org.apache.spark.rdd.RDD

import scala.reflect.ClassTag

/*----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/

object SparkParser
{
    def parseJson[T: ClassTag]( input: RDD[String], schematizer: JsonNode => T ): RDD[T] =
    {
        val mapper = new ObjectMapper

        input.map( ( string: String ) => schematizer( mapper.readTree( string ) ) )
    }
}

/*----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------*/
